src/main/java/net/sf/jabref/model/database/BibDatabase.java

```diff
@@ -173,15 +171,14 @@ public synchronized boolean insertEntry(BibEntry entry) throws KeyCollisionExcep
     /**
      * Removes the given entry.
      */
-    public synchronized void removeEntry(BibEntry oldValue) {
-        if (oldValue == null) {
+    public synchronized void removeEntry(BibEntry toBeDeleted) {
+        if (toBeDeleted == null) {
             return;
         }
 
-        entries.remove(oldValue.getId());
-
```

Comment:
Ahh, I just fixed that (but worse, just removed the getId()-part). Will removed that commit from my PR.